src/plots/cartesian/axes.js

```diff
@@ -1203,6 +1210,46 @@ axes.calcTicks = function calcTicks(ax, opts) {
     return ticksOut;
 };
 
+function syncTicks(ax) {
+    // get the overlaying axis
+    var baseAxis = ax._mainAxis;
+
+    var ticksOut = [];
+    if(baseAxis && baseAxis._vals) {
+        for(var i = 0; i < baseAxis._vals.length; i++) {
+            // get the position of the every tick
+            var pos = baseAxis.l2p(baseAxis._vals[i].x);
+
+            // get the tick for the current axis based on position
+            var vali = ax.p2l(pos);
+            var val1 = ax.p2l(pos - 0.5);
+            var val2 = ax.p2l(pos + 0.5);
+            var d = 1 + Math.round(Math.log10(Math.abs(val2 - val1)));
+            var e = Math.pow(10, -d);
+            var valR = Math.round(vali * e) / e;
+            var objR = axes.tickText(ax, valR);
+            var obj = axes.tickText(ax, vali);
+            obj.text = objR.text;
+
+            // assign minor ticks
+            if(baseAxis._vals[i].minor) {
+                obj.minor = true;
+                obj.text = '';
+            }
+
+            ticksOut.push(obj);
+        }
+    }
+
+    if(ax.rangebreaks) {
```

Comment:
We have this exact block in `arrayTicks`.
It would be nice to put it in a function and call it from both places.